### ISSUE TITLE
Ubuntu 22.04 LTS issue <built-in function load_binary> returned NULL without setting an exception

### ISSUE THREAD

**Issue by krim404** [pytorch/pytorch#98707]
### 🐛 Describe the bug

Greetings,

I was directed to this repository as I am encountering an issue with PyTorch. Specifically, I am experiencing an error with loading triton when attempting to run the software with stable diffusion and dreambooth addon on a newly installed KUbuntu 22.04 operating system on an AMD CPU with an AMD GPU (6950XT) and rocm 5.4.2 (installed via official deb package)

Regrettably, I do not possess a high level of proficiency in Python, and as such, I can only describe the steps that I have taken on a freshly installed machine.

The system is set up by cloning the repository https://github.com/AUTOMATIC1111/stable-diffusion-webui with torch rocm via `pip install torch torchvision --extra-index-url https://download.pytorch.org/whl/rocm5.4.2` in venv as well as dreambooth extension. No further modifications are done. In the training of the model it stops with the following error message:

```
Steps:   0%|                                                                                                                                                                                          | 0/600 [00:00<?, ?it/s]Traceback (most recent call last):
  File "/home/krim/GIT/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/ui_functions.py", line 727, in start_training
    result = main(class_gen_method=class_gen_method)
  File "/home/krim/GIT/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/train_dreambooth.py", line 1371, in main
    return inner_loop()
  File "/home/krim/GIT/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/memory.py", line 119, in decorator
    return function(batch_size, grad_size, prof, *args, **kwargs)
  File "/home/krim/GIT/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/train_dreambooth.py", line 1169, in inner_loop
    noise_pred = unet(
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/accelerate/utils/operations.py", line 495, in __call__
    return convert_to_fp32(self.model_forward(*args, **kwargs))
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/amp/autocast_mode.py", line 14, in decorate_autocast
    return func(*args, **kwargs)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 82, in forward
    return self.dynamo_ctx(self._orig_mod.forward)(*args, **kwargs)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 209, in _fn
    return fn(*args, **kwargs)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/diffusers/models/unet_2d_condition.py", line 556, in forward
    t_emb = self.time_proj(timesteps)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/diffusers/models/embeddings.py", line 222, in forward
    def forward(self, timesteps):
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 209, in _fn
    return fn(*args, **kwargs)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 2819, in forward
    return compiled_fn(full_args)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 1222, in g
    return f(*args)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 1898, in runtime_wrapper
    all_outs = call_func_with_args(
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 1247, in call_func_with_args
    out = normalize_as_list(f(args))
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 248, in run
    return model(new_inputs)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 265, in run
    compiled_fn = cudagraphify_impl(model, new_inputs, static_input_idxs)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 320, in cudagraphify_impl
    model(list(static_inputs))
  File "/tmp/torchinductor_krim/xq/cxqqlwutwzuplluoktmemj63w363iojzltjm3s3avxbjioaget6s.py", line 113, in call
    triton__0.run(arg0_1, buf0, buf1, 160, grid=grid(160), stream=stream0)
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/_inductor/triton_ops/autotune.py", line 190, in run
    result = launcher(
  File "<string>", line 6, in launcher
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/triton/compiler.py", line 1944, in __getattribute__
    self._init_handles()
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/triton/compiler.py", line 1930, in _init_handles
    mod, func, n_regs, n_spills = hip_utils.load_binary(self.metadata["name"], self.asm["hsaco_path"], self.shared, device)
SystemError: <built-in function load_binary> returned NULL without setting an exception
```

See https://github.com/d8ahazard/sd_dreambooth_extension/issues/1174

On Reddit, I stumbled upon only one other individual who reported facing the same issue. Unfortunately, they did not provide any insights on how to even begin debugging this issue. 

Are there any suggestions or ideas that anyone could offer how to start debugging it?

### Versions

```
 
Collecting environment information...
/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/cuda/__init__.py:546: UserWarning: Can't initialize NVML
  warnings.warn("Can't initialize NVML")
PyTorch version: 2.0.0+rocm5.4.2
Is debug build: False
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: 5.4.22803-474e8620

OS: Ubuntu 22.04.2 LTS (x86_64)
GCC version: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Clang version: Could not collect
CMake version: version 3.26.1
Libc version: glibc-2.35

Python version: 3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0] (64-bit runtime)
Python platform: Linux-6.2.10-060210-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: AMD Radeon RX 6950 XT
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: 5.4.22803
MIOpen runtime version: 2.19.0
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   48 bits physical, 48 bits virtual
Byte Order:                      Little Endian
CPU(s):                          24
On-line CPU(s) list:             0-23
Vendor ID:                       AuthenticAMD
Model name:                      AMD Ryzen 9 5900X 12-Core Processor
CPU family:                      25
Model:                           33
Thread(s) per core:              2
Core(s) per socket:              12
Socket(s):                       1
Stepping:                        0
Frequency boost:                 enabled
CPU max MHz:                     4950,1948
CPU min MHz:                     2200,0000
BogoMIPS:                        7386.11
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm
Virtualization:                  AMD-V
L1d cache:                       384 KiB (12 instances)
L1i cache:                       384 KiB (12 instances)
L2 cache:                        6 MiB (12 instances)
L3 cache:                        64 MiB (2 instances)
NUMA node(s):                    1
NUMA node0 CPU(s):               0-23
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP always-on, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] lion-pytorch==0.0.7
[pip3] numpy==1.23.3
[pip3] open-clip-torch==2.7.0
[pip3] pytorch-lightning==1.9.4
[pip3] pytorch-triton-rocm==2.0.1
[pip3] torch==2.0.0+rocm5.4.2
[pip3] torchdiffeq==0.2.3
[pip3] torchmetrics==0.11.4
[pip3] torchsde==0.2.5
[pip3] torchvision==0.15.1+rocm5.4.2
[conda] Could not collect

```
cc @ezyang @gchanan @zou3519 @kadeng @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @msaroufim @bdhirsh @anijain2305 @chauhang @soumith @wconstab @ngimel @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

**Comment by krim404** [comment]
It's possible that the issue isn't specific to Rocm because another person who was using Ubuntu 22.04 and an NVidia card reported the same problem on the other ticket in the sd_dreambooth repo.

**Comment by jithunnair-amd** [comment]
```
  File "/home/krim/GIT/stable-diffusion-webui/venv/lib/python3.10/site-packages/triton/compiler.py", line 1930, in _init_handles
    mod, func, n_regs, n_spills = hip_utils.load_binary(self.metadata["name"], self.asm["hsaco_path"], self.shared, device)
SystemError: <built-in function load_binary> returned NULL without setting an exception 
```

@rsanthanam-amd Could the above error message indicate an unsupported gfx target?

**Comment by rsanthanam-amd** [comment]
@jithunnair-amd The OP stated that they used the 6950XT which is Navi21. This should be supported so it is likely that a different issue is causing this.

**Comment by wconstab** [comment]
@rsanthanam-amd i'm marking this as triaged from our end and assigning it to 'AMD' generally- i'll also assign it to you you personally just for accountability but if you could reassign it to the right person that'd be great.

**Comment by seungrokjung** [comment]
As the load_binary function was not able to read data when all of arguments for load_binary are correct, I guess the device needs some time to prepare for sending the results. I fixed the bug like this was: FYI. 

**1930             import time
1931             time.sleep(0.001)**
1932             mod, func, n_regs, n_spills = hip_utils.load_binary(self.metadata["name"], self.asm["hsaco_path"], self.shared, device)

**Comment by seungrokjung** [comment]
I'm not 100% sure if solves all cases. But I tested on mi210 and rx6900xt and worked well. Try to remove everything under /tmp/* and set ROCM_PATH environment and run. 
I guess the initial unsuccessful run (maybe not setting /opt/rocm path in the beginning) created garbage data under /tmp/* during compilation and they hinder the further compilation. 
==
**export ROCM_PATH=/opt/rocm
rm -rf /tmp/***
==


root@050d440bacfb:/GPTQ# export ROCM_PATH=/opt/rocm
root@050d440bacfb:/GPTQ# python3 run_genai_sd.py 
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["id2label"]` will be overriden.
/usr/local/lib/python3.10/dist-packages/transformers/models/clip/feature_extraction_clip.py:28: FutureWarning: The class CLIPFeatureExtractor is deprecated and will be removed in version 5 of Transformers. Please use CLIPImageProcessor instead.
  warnings.warn(
[INFO] running mode: pth2.0 compilation
  0%|                                                                                                                                                                                                                                                                                                                                                                | 0/32 [00:01<?, ?it/s]
Traceback (most recent call last):
  File "/GPTQ/run_genai_sd.py", line 49, in <module>
    run_sd(k, itr, comp)
  File "/GPTQ/run_genai_sd.py", line 36, in run_sd
    image = pipe(prompt, num_inference_steps = steps, num_images_per_prompt = batch_size, height=height, width=width).images
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 676, in __call__
    noise_pred = self.unet(
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/_dynamo/eval_frame.py", line 82, in forward
    return self.dynamo_ctx(self._orig_mod.forward)(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/_dynamo/eval_frame.py", line 209, in _fn
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/diffusers/models/unet_2d_condition.py", line 659, in forward
    t_emb = self.time_proj(timesteps)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/diffusers/models/embeddings.py", line 222, in forward
    def forward(self, timesteps):
  File "/usr/local/lib/python3.10/dist-packages/torch/_dynamo/eval_frame.py", line 209, in _fn
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/_functorch/aot_autograd.py", line 2819, in forward
    return compiled_fn(full_args)
  File "/usr/local/lib/python3.10/dist-packages/torch/_functorch/aot_autograd.py", line 1222, in g
    return f(*args)
  File "/usr/local/lib/python3.10/dist-packages/torch/_functorch/aot_autograd.py", line 1898, in runtime_wrapper
    all_outs = call_func_with_args(
  File "/usr/local/lib/python3.10/dist-packages/torch/_functorch/aot_autograd.py", line 1247, in call_func_with_args
    out = normalize_as_list(f(args))
  File "/usr/local/lib/python3.10/dist-packages/torch/_inductor/compile_fx.py", line 248, in run
    return model(new_inputs)
  File "/tmp/torchinductor_seungrok/b4/cb4wgmozs4bqzsoemk2mnsij7oypadaj3znkbzxbgpyxre3st3cy.py", line 116, in call
    triton__0.run(arg0_1, buf0, buf1, 320, grid=grid(320), stream=stream0)
  File "/usr/local/lib/python3.10/dist-packages/torch/_inductor/triton_ops/autotune.py", line 190, in run
    result = launcher(
  File "<string>", line 6, in launcher
  File "/usr/local/lib/python3.10/dist-packages/triton/compiler.py", line 1944, in __getattribute__
    self._init_handles()
  File "/usr/local/lib/python3.10/dist-packages/triton/compiler.py", line 1930, in _init_handles
    mod, func, n_regs, n_spills = hip_utils.load_binary(self.metadata["name"], self.asm["hsaco_path"], self.shared, device)
SystemError: <built-in function load_binary> returned NULL without setting an exception
root@050d440bacfb:/GPTQ# rm -rf /tmp/*
root@050d440bacfb:/GPTQ# python3 run_genai_sd.py 
/usr/local/lib/python3.10/dist-packages/transformers/models/clip/feature_extraction_clip.py:28: FutureWarning: The class CLIPFeatureExtractor is deprecated and will be removed in version 5 of Transformers. Please use CLIPImageProcessor instead.
  warnings.warn(
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["id2label"]` will be overriden.
[INFO] running mode: pth2.0 compilation
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 32/32 [01:06<00:00,  2.06s/it]
/usr/local/lib/python3.10/dist-packages/diffusers/pipelines/pipeline_utils.py:1363: RuntimeWarning: invalid value encountered in cast
  images = (images * 255).round().astype("uint8")
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 32/32 [00:03<00:00, 10.62it/s]
3.11433s for the calculation
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 32/32 [00:03<00:00, 10.62it/s]
3.10763s for the calculation
/usr/local/lib/python3.10/dist-packages/transformers/models/clip/feature_extraction_clip.py:28: FutureWarning: The class CLIPFeatureExtractor is deprecated and will be removed in version 5 of Transformers. Please use CLIPImageProcessor instead.
  warnings.warn(
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["id2label"]` will be overriden.
[INFO] running mode: pth2.0 eager
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 32/32 [00:03<00:00,  9.83it/s]
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 32/32 [00:03<00:00,  9.89it/s]
3.33001s for the calculation
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 32/32 [00:03<00:00,  9.85it/s]
3.34668s for the calculation

**Comment by anicolson** [comment]
I am also getting this issue with an MI250X.

**Comment by anicolson** [comment]
> I'm not 100% sure if this solves all cases. But I tested on mi210 and rx6900xt and worked well. Try to remove everything under /tmp and set ROCM_PATH environment and run. I guess the initial unsuccessful run (maybe not setting /opt/rocm path in the beginning) created garbage data under /tmp during compilation and they hinder further compilation.
> 
> export ROCM_PATH=/opt/rocm rm -rf /tmp/*
> 
> ![Screenshot from 2023-05-03 21-42-24](https://user-images.githubusercontent.com/85979992/235919197-7e929251-aea8-4484-a43d-49f84df82222.png)

I am also getting this issue with an MI250X.

After trying this, I get:
```
ld.lld execute fail: 
Executable "/llvm/bin/ld.lld" doesn't exist!
-1
```

**Comment by dllehr-amd** [comment]
hi @anicolson,  That issue should have been resolved already.  The root issue in this case is that ROCM_PATH isn't being set.  

We've already resolved this to allow future whl's of triton to include the `ld.lld` with it, as well as attempting to look in default rocm install locations if `ROCM_PATH` isn't set.  

It seems like you may have an older version of PyTorch/Triton on ROCm?  would you be able to send me the levels of both pytorch/triton/pytorch_triton_rocm(if applicable)?

We can get this sorted out for you quickly.

**Comment by anicolson** [comment]
Hi @dllehr-amd,

I am using pytorch version: 2.0.1+rocm5.4.2 (rocm 5.4.3 is available on the cluster I am using).

Thanks!

**Comment by jithunnair-amd** [comment]
@anicolson This issue should also be fixed in the nightly wheels of PyTorch, but they use a newer ROCm now:
```
pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/rocm5.5
```

**Comment by 324170455** [comment]
> 我还在使用MI250X时遇到这个问题。do you have amd mi250x linux or windows driver?

**Comment by netw0rkf10w** [comment]
This problem occurs to me very frequently on a server with MI250x devices. Has anyone come up with a resolution please?

**Comment by tugsbayasgalan** [comment]
@rsanthanam-amd any updates?

**Comment by britishtiger** [comment]
Same problem on python 3.10, pytorch 2.2.2+rocm5.7.
Setting `ROCM_PATH` env didn't work for me, but I found a very strange workaround. Trying to find cause of the problem I added a print statement (`print(self.device_type)`) before `if self.device_type in ["cuda"]:` line in `_init_handles` function in `triton/compiler/compiler.py` and this action somehow resolved the error. When I remove this line, the same error returns again. This behavior is beyond my understanding of python, but I hope this information can be helpful

**Comment by hongxiayang** [comment]
Can you folks try to use nightly, to see if problem still there? 
```
pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/rocm6.0
```

**Comment by pruthvistony** [comment]
Suggest to move to ROCm 6.0 atleast, since they was ROCm lib path restructuring was enforced to path /opt/rocm/lib all libs were moved to this folder. This solved many flaky issues and standardized the lib paths.

**Comment by britishtiger** [comment]
> Same problem on python 3.10, pytorch 2.2.2+rocm5.7. Setting `ROCM_PATH` env didn't work for me, but I found a very strange workaround. Trying to find cause of the problem I added a print statement (`print(self.device_type)`) before `if self.device_type in ["cuda"]:` line in `_init_handles` function in `triton/compiler/compiler.py` and this action somehow resolved the error. When I remove this line, the same error returns again. This behavior is beyond my understanding of python, but I hope this information can be helpful

Well, the error returned again. I commented out the statement I added and it works again. My uneducated guess that adding print messes with execution flow somehow (race condition)? In any case, hope this information can be helpful for the devs

**Comment by hegemanjwh3** [comment]
I have run into this error as well.  I am suspicious that it is related to, or otherwise affected by, the "triton cache" under ~/.triton/cache.

My environment:
- Ubuntu 24.04
- ROCm 6.1.2
- Navi33 GPU
- Python 3.11
- PyTorch 2.3.1

Ensuring that ROCM_PATH was set didn't work for me either; nor did clearing amd_triton_kernel-... stuff out of /tmp.  However, I had success with @tsukidoke1 's observation of adding a print command to compiler.py.  (See tsukidoke1's comment above for specifics.)  But -- I experienced very ephemeral behavior; sometimes I had to actually *change* the print statement to relieve the error.  I also noticed that the error could appear, but then be "resolved" by changing some seemingly-irrelevant line in my source file (rather than within the compiler).  This suggests to me that this issue is somehow related to some caching somewhere.  I tried clearing all __pycache__ dirs under my triton install, but this didn't help.  Then I stumbled on ~/.triton/cache/.  Clearing out this directory seemed to me to alleviate the error -- for instance I was able to remove my print commands from compiler.py after clearing this cache, and I wouldn't immediately get the error again.

So, tentatively my recommendation is to simply clear out the triton cache under ~/.triton/cache/ when this error appears.  This has been the most reliable fix I have observed (but I only discovered this today, so we'll see if it holds up...)
